src/qttools/nevp/utils.py

```diff
@@ -0,0 +1,48 @@
+from qttools.utils.gpu_utils import xp
+
+
+def contour_quadrature(radius: float, num_quad_points: int):
+    """Computes quadrature points and weights for contour integration.
+
+    This function computes the quadrature points and weights for
+    contour integration on a circular contour of a given radius using
+    the trapezoidal rule.
+
+    Parameters
+    ----------
+    radius : float
+        The radius of the circular contour.
+    num_quad_points : int
+        The number of quadrature points to use.
+
+    Returns
+    -------
+    z, w : tuple[xp.ndarray, xp.ndarray, xp.ndarray, xp.ndarray]
+        The quadrature points and weights.
+
+    """
+    zeta = xp.arange(num_quad_points) + 1
+    z = radius * xp.exp(2j * xp.pi * (zeta + 0.5) / num_quad_points)
+
+    w = xp.ones(num_quad_points) / num_quad_points
+
+    # Reshape to broadcast over batch dimension.
+    z = z.reshape(1, -1, 1, 1)
+    w = w.reshape(1, -1, 1, 1)
+
+    return z, w
+
+
+def operator_inverse(
```

Comment:
Need documentation for this function, don't know what it's doing

Reply:
This computes the $T^{-1}(z)$ that appears in the contour integral in subspace methods. Adding a docstring :+1:

Reply:
perfect, thanks